unity-renderer/Assets/DCLServices/WearablesCatalogService/LambdasWearablesCatalogService.cs

```diff
@@ -0,0 +1,285 @@
+        private readonly Dictionary<(string, string), LambdaResponsePagePointer<WearableResponse>> thirdPartyCollectionPagePointers = new ();
+        private readonly Dictionary<string, UniTaskCompletionSource<WearableItem>> awaitingWearableTasks = new ();
+        private readonly List<string> pendingWearablesToRequest = new ();
+
+        public LambdasWearablesCatalogService(BaseDictionary<string, WearableItem> wearablesCatalog)
+        {
+            WearablesCatalog = wearablesCatalog;
+        }
+
+        public void Initialize()
+        {
+            serviceCts = serviceCts.SafeRestart();
+
+            // All the requests happened during the same frames interval are sent together
+            CheckForSendingPendingRequestsAsync(serviceCts.Token).Forget();
+
+            // Check unused wearables (to be removed from our catalog) only every [TIME_TO_CHECK_FOR_UNUSED_WEARABLES] seconds
+            CheckForUnusedWearablesAsync(serviceCts.Token).Forget();
+        }
+
+        public void Dispose()
+        {
+            serviceCts.SafeCancelAndDispose();
+            serviceCts = null;
+            Clear();
+        }
+
+        public async UniTask<WearableItem[]> RequestOwnedWearablesAsync(string userId, int pageNumber, int pageSize, CancellationToken ct)
+        {
+            if (!ownerWearablesPagePointers.ContainsKey(userId))
+            {
+                ownerWearablesPagePointers[userId] = new (
+                    WEARABLES_BY_OWNER_END_POINT.Replace("{userId}", userId),
+                    pageSize, ct, this);
+            }
+
+            var pageResponse = await ownerWearablesPagePointers[userId].GetPageAsync(pageNumber, ct);
+            AddWearablesToCatalog(pageResponse.response.wearables.ToArray());
+
+            return pageResponse.response.wearables.ToArray();
+        }
+
+        public async UniTask<WearableItem[]> RequestBaseWearablesAsync(CancellationToken ct)
+        {
+            if (!wearablesByCollectionPagePointers.ContainsKey(BASE_WEARABLES_COLLECTION_ID))
+            {
+                wearablesByCollectionPagePointers[BASE_WEARABLES_COLLECTION_ID] = new (
+                    WEARABLES_BY_COLLECTION_END_POINT.Replace("{collectionId}", BASE_WEARABLES_COLLECTION_ID),
+                    1, ct, this);
+            }
+
+            var pageResponse = await wearablesByCollectionPagePointers[BASE_WEARABLES_COLLECTION_ID].GetPageAsync(1, ct);
+            AddWearablesToCatalog(pageResponse.response.wearables.ToArray());
+
+            return pageResponse.response.wearables.ToArray();
+        }
+
+        public async UniTask<WearableItem[]> RequestThirdPartyWearablesByCollectionAsync(string userId, string collectionId, CancellationToken ct)
+        {
+            if (!thirdPartyCollectionPagePointers.ContainsKey((userId, collectionId)))
+            {
+                thirdPartyCollectionPagePointers[(userId, collectionId)] = new (
```

Comment:
Hmmm... it's true. Any suggestion for a caching strategy here? The current `CatalogController` is not currently caching requests. In order to not to go too far off plan, maybe we should focus this refactor to keep the current behavior as it is and think on caching improvements for future steps. Wdyt?

Reply:
1. You can't leave it as is because it will not function properly for users, from the users perspective it's just a bug
2. Caching is a part of the task, and it's already here but just implemented not properly
3. With the API chosen you can't remove caching as it will lead to recreation of `PagePointer` every time a page is requested, even within the same view navigation. If you chose to expose `PagePointer` instead you could do simple caching on the view level from `Open` to `Close` (Disposing `PagePointer` on view close, and reusing the results while the view is open)
4. If I were to implement it I would introduce a generic cache with a lifetime for its entries, e.g. `PagePointer` is kept no longer than X seconds, and then on a subsequent request if the lifetime is exceeded it's recreated. We could go one step further and introduce tiers via Feature Flags Variants to A/B test UX. I would also limit the capacity for such cache according to LRU principle as it can become a problem.

Reply:
Finally, I have followed this approach to manage the cached pages, let's see what you think:

I have added a new boolean param `cleanCachedPages` in the signature of both paginated methods:
```
UniTask<IReadOnlyList<WearableItem>> RequestOwnedWearablesAsync(string userId, int pageNumber, int pageSize, bool cleanCachedPages, CancellationToken ct);
UniTask<IReadOnlyList<WearableItem>> RequestThirdPartyWearablesByCollectionAsync(string userId, string collectionId, int pageNumber, int pageSize, bool cleanCachedPages, CancellationToken ct);
```
Depending on that `cleanCachedPages` param, we will dispose the existing `PagePointer` and re-create it again. In this way we let the consumers to decide if they want to force the cleaning of the already cached pages or not before doing each request.

For example: When we open the Passport, we can request the first page of wearables using `cleanCachedPages = true`, but we can keep it as `cleanCachedPages = false` while we are navigating inside the open passport.

You can see the implementation here: https://github.com/decentraland/unity-renderer/pull/4084/commits/404bd835734608579c79f81b8c072d4f14f3c613

@mikhail-dcl wdyt?